2/answers.txt

```diff
@@ -0,0 +1,231 @@
+Q1: Why is password authentication inferior to public key authentication?
+
+Because Key based authentication is computer generated, rather than human generated, and typically 4096 Bits in length
+less prone to being brute forced, acquired by subterfuge or guessed than a traditional password. 
+
+Communications between two system using a key pair method is only possible to decrypted by owners of the key pair.
+Furthermore a private key can typically be further encrypted with a standard password to increase safety.
+
+Q2: Why is VSCode a good choice for a text editor and how does it differ from a fully-fledged IDE?
+
+Visual Studio and other IDEs are intended as entire application or project construction platforms, whereas VSCode is a
+more of a build and debug text editor. It lacks some functionality that VS has however it is fast, and well designed 
+for cloud applications.
```

Comment:
VS Code also has use as an Ops tool, with native terminal, remote SSH and remote docker support, which is something currently lacking in Visual Studio.
For example, in VS Code you are able to remote into a server, open a workspace inside a docker container running in the server, use the terminal inside the container and also have visual access to the filesystem inside the container.

It's also a great choice for web development as it has best-in-class support for HTML, CSS, Javascript/Typescript, JSON etc.

I often use VS Code to remote into a VM and use it as a dev environment. For example a local Hyper-V VM or a cloud instance. the benefit of this is that it's easy to standardise the environment. Get a new PC? install VS Code and move the VM over, done.

Visual Studio is surely the best for C# development however, as all its tooling has been designed for that single purpose.